### ISSUE TITLE
[v2.4 web.browser.legacy] update to code-js v3.16.0 introduces iterator issue

### ISSUE THREAD

**Issue by brucejo75** [meteor/meteor#11662]
For legacy builds, Babel is throwing errors when trying to convert iterators into arrays.  It shows up when I try to use the spread operator.  This may be a recall class bug.

(Took me a while to encounter it because it only showed up in legacy builds, which for me is my `cordova` build)

## Environment
- meteor@2.4 w/core-js@3.16.0
- Ubuntu 20.04.3 LTS
- @babel/runtime@7.15.4

## Repro
I have a repo for this repro at https://github.com/brucejo75/spread-operator
But the instructions are pretty simple too...

<details>

<summary>Repro steps</summary>

```bash
meteor create test
cd test
```

Change `main.jsx` code to:

```javascript
Meteor.startup(() => {
  render(<App/>, document.getElementById('react-target'));
  let nodes = document.querySelectorAll("p");
  console.log('nodes[Symbol.iterator]: ', nodes[Symbol.iterator]);
  foobar(...nodes);
});

function foobar(...args) {
  console.log(args.join());
}
```

build and run a legacy only build
```bash
meteor run --exclude-archs=web.browser
```

connect to [https://localhost:3000](https://localhost:3000)
</details>

<details>

<summary>Error to console</summary>

```console
main.jsx:9 nodes[Symbol.iterator]:  undefined
VM2414 modules.js:1257 Uncaught TypeError: Invalid attempt to spread non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
    at _nonIterableSpread (VM2414 modules.js:1257)
    at _toConsumableArray (VM2414 modules.js:1162)
    at main.jsx:10
    at maybeReady (meteor.js?hash=b9ec8cf25b6fc794ae6b825f30e06c3c35c50e7c:943)
    at HTMLDocument.loadingCompleted (meteor.js?hash=b9ec8cf25b6fc794ae6b825f30e06c3c35c50e7c:955)
_nonIterableSpread @ VM2414 modules.js:1257
_toConsumableArray @ VM2414 modules.js:1162
(anonymous) @ main.jsx:10
maybeReady @ meteor.js?hash=b9ec8cf25b6fc794ae6b825f30e06c3c35c50e7c:943
loadingCompleted @ meteor.js?hash=b9ec8cf25b6fc794ae6b825f30e06c3c35c50e7c:955
```

</details>

## Workaround
For non-array iterables:
use [`apply`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) instead of spread operator.
use [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) with your spread operator.

```
func.apply(func, iterable);
-or-
func(...Array.from(iterable));
```

## Rollback [core-js](https://github.com/zloirock/core-js)

Rolling back to `core-js@3.15.2` from [`core-js@3.16.0`](https://github.com/meteor/meteor/blob/release-2.4/packages/ecmascript-runtime-client/package.js#L10) will make the issue go away.  This looks like a difference in defining iterables in `core-js`.

**Note**: I also tried the latest version `core-js@3.18.1` and it is broken too.

## Observations
- A NodeList is definitely supposed to be iterable.
- [iterableToArray](https://github.com/babel/babel/blob/main/packages/babel-runtime/helpers/iterableToArray.js#L2) is what is failing.  `iter[Symbol.iterator[` should return a value.
- Not sure if this is an issue with [`core-js`](https://github.com/zloirock/core-js) or meteor's interaction with babel.

**Comment by filipenevola** [comment]
It seems like the rollback is the way to go.

We also should include new test cases including Iterators being converted to arrays with spread operator to avoid this regression again in the future.

**Comment by StorytellerCZ** [comment]
@brucejo75 first of all thank you for an amazingly written bug report. I agree with @filipenevola that rolling back to `core-js@3.15.2` is the fastest approach and adding regressions test for this. Since it is Hacktoberfest I give you the option to create the PR with the rollback & tests first if you want, let me know if you need any help with that or if you want me to take care of it.

**Comment by brucejo75** [comment]
@filipenevola, @StorytellerCZ,

I agree rollback is the first right thing to do.  I will take a shot at creating a PR. 

But, it seems to me this should also be looked into more closely?

I have no experience with core-js/babel or meteor's interactions with it.  Otherwise I would have dug some more.  Is there someone who is familiar with the meteor/babel interactions and could dig into that?

Either, there is an issue with Meteor or an issue needs to be raised with `core-js`.

Or if someone can point me in some kind of direction, I could take a closer look.

**Comment by StorytellerCZ** [comment]
Looking at the code, core-js is used for polyfills, so maybe there is something wrong with compat checking or something changed in the polyfills.

**Comment by StorytellerCZ** [comment]
`core-js@3.16.0` did some changes to arrays, but nothing that pops out to me...
https://github.com/zloirock/core-js/blob/master/CHANGELOG.md#3160---20210730

**Comment by StorytellerCZ** [comment]
@brucejo75 please submit a quick PR for the rollback and then we can take care of adding the tests.

**Comment by brucejo75** [comment]
@StorytellerCZ, PR #11665 available.  Will work on the tests...

**Comment by brucejo75** [comment]
There are a couple of tests for iterators that should have failed, but they do not:

[Map spread](https://github.com/meteor/meteor/blob/release-2.4/packages/ecmascript/runtime-tests.js#L275)
[Set spread](https://github.com/meteor/meteor/blob/release-2.4/packages/ecmascript/runtime-tests.js#L289-L297)

Turns out this bug is specific to the [`NodeLlist`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) class.  And `NodeList` is not instantiable.

I get this error when I try to instantiate: `Uncaught TypeError: Illegal constructor`

To test `NodeList` I need the DOM present to get a `NodeList`.  Is this possible with `Tinytest`?

**Comment by StorytellerCZ** [comment]
Fix published in `ecmascript-runtime-client@0.12.1`
@brucejo75 thanks for the tests analysis. That would explain why it slip through. As for the tests, I have seen DOM used in `less` tinytests. Take a look on those, but I'm not sure if that is going to work here. I will look into this as well.
